pkg/crc/preflight/preflight_checks_common.go

```diff
@@ -158,7 +161,57 @@ func isDaemonRunningWithReleasedVersion() bool {
 	}
 	if ver.CrcVersion != version.GetCRCVersion() {
 		logging.Debugf("Daemon is running with %s version but binary version is %s", ver.CrcVersion, version.GetCRCVersion())
+		if err := killExistingDaemonProcess(); err != nil {
+			logging.Debugf("Failed to kill the existing crc daemon process: %v", err)
+		}
 		return false
 	}
 	return true
 }
+
+func killExistingDaemonProcess() error {
+	crcProcessName := "crc"
+	if runtime.GOOS == "windows" {
+		crcProcessName = "crc.exe"
+	}
+	processes, err := process.Processes()
+	if err != nil {
+		return err
+	}
+	for _, process := range processes {
+		name, err := process.Name()
+		if err != nil {
+			return err
+		}
+		if name == crcProcessName {
+			cmdLine, err := process.CmdlineSlice()
+			if err != nil {
+				return err
+			}
+			if isDaemonProcess(cmdLine) {
+				logging.Debugf("Got the pid for %s : %d", name, process.Pid)
+				if err := process.Kill(); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// isDaemonProcess return true if following conditions met
```

Comment:
if any of the following conditions are met? if all of the following conditions are met? if none of the following conditions are met?

Reply:
@cfergeau updated to `any of the following`.